engine/access/state_stream/backend/backend.go

```diff
@@ -212,3 +217,15 @@ func (b *StateStreamBackend) getStartHeight(startBlockID flow.Identifier, startH
 func (b *StateStreamBackend) setHighestHeight(height uint64) bool {
 	return b.highestHeight.Set(height)
 }
+
+// GetRegisterValues returns the register values for the given register IDs at the given block height.
+func (b *StateStreamBackend) GetRegisterValues(ids flow.RegisterIDs, height uint64) ([]flow.RegisterValue, error) {
+	values, err := b.registers.RegisterValues(ids, height)
+	if errors.Is(err, storage.ErrHeightNotIndexed) {
+		return nil, status.Errorf(codes.OutOfRange, "register values for block %d is not available yet", height)
```

Comment:
it may never become available if it's below the root height
```suggestion
		return nil, status.Errorf(codes.OutOfRange, "register values for block %d is not available", height)
```